tools/license/get_dep_licenses.go

```diff
@@ -184,12 +254,61 @@ func main() {
 		linfo.path = lp
 		linfo.url = pathToURL(lp)
 		licenses = append(licenses, linfo)
+		ltypeStr := linfo.licenseTypeString
 		if linfo.exact {
-			licenseTypes[linfo.licenseTypeString] = append(licenseTypes[linfo.licenseTypeString], p)
+			licenseTypes[ltypeStr] = append(licenseTypes[ltypeStr], p)
 			exact = append(exact, linfo)
 		} else {
 			inexact = append(inexact, linfo)
 		}
+
+		switch {
+		case approvedLicenses[ltypeStr]:
+		case reciprocalLicenses[ltypeStr]:
+			reciprocalList = append(reciprocalList, linfo.packageName)
+		case restrictedLicenses[ltypeStr]:
+			restrictedList = append(restrictedList, linfo.packageName)
+		case ltypeStr == "":
+			missingList = append(missingList, linfo.packageName)
+		default:
+			if !knownUnknownLicenses[linfo.packageName] {
```

Comment:
are the known unknown licenses OK?

Reply:
which comment? I looked though things around unknown but I am not sure I understand them (not the code, but the known unknowns. =)

Were they previously checked and verified? If so, does it mean they are OK?
If not, how are they different from the real unknowns?

Reply:
You mean the unknown unknowns? :-)
The idea is that the licenses are now in a fully reviewed state. We had to ok some manually for various reasons (e.g. they were not embedded but had a link or had some atypical license text that legal had to review). I'm now calling these the "known unknowns" because they are unknown to the licensee tool (which will complain), but known to us. Let me see if I can make the variable text a little clearer.